File: Backend/src/main/java/de/ali/shitpostbot/Coordinate/model/Coordinate.java
package de.ali.shitpostbot.Coordinate.model;

import de.ali.shitpostbot.Template.model.Template;
import de.ali.shitpostbot.shared.model.BaseEntity;
import lombok.EqualsAndHashCode;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

import javax.persistence.Entity;
import javax.persistence.JoinColumn;
import javax.persistence.ManyToOne;
import java.io.Serializable;

/**
 * Ideen zur Handhabung von Koordinaten
 * Stehen in N:1 Relation zu Templates
 * Sollen einem Template Koordinaten hinzugefügt oder entfernt werden, so geschieht
 * dies im Frontend
 * Dann werden da auch die Previews nochmal geladen
 *
 * Koordinaten werden genutzt, um Kasten um bereiche zu malen wo die crops stattfinden
 */

@Entity
@Getter
@Setter
@NoArgsConstructor
public class Coordinate extends BaseEntity implements Serializable {

    @ManyToOne
    @JoinColumn
    @EqualsAndHashCode.Exclude
    private Template reference;

    /**
     * Koordinate LINKS
     */
    private int x1;
    /**
     * Koordinate RECHTS
     */
    private int x2;
    /**
     * Koordinate OBEN
     */
    private int y1;
    /**
     * Koordinate UNTEN
     */
    private int y2;


}

File: Frontend/src/shared/coordinate/model/Coordinate.ts
import {BaseEntity} from '../../models/BaseEntity';
import {Template} from '../../template/model/Template';

export interface Coordinate extends BaseEntity {
  reference: Template;
  x1: number;
  x2: number;
  y1: number;
  y2: number;
}

/*export class Coordinate extends BaseEntity {
  public reference: Template;
  public x1: number;
  public x2: number;
  public y1: number;
  public y2: number;

  /*public get reference(): Template {
    return this._reference;
  }
  public set reference(reference: Template) {
    this._reference = reference;
  }
  public get x1(): number {
    return this._x1;
  }
  public set x1(x1: number) {
   this._x1 = x1;
  }
  public get x2(): number {
    return this._x2;
  }
  public set x2(x2: number) {
    this._x2 = x2;
  }
  public get y1(): number {
    return this._y1;
  }
  public set y1(y1: number) {
    this._y1 = y1;
  }
  public get y2(): number {
    return this._y1;
  }
  public set y2(y2: number) {
    this._y2 = y2;
  }
}*/
